app/routes.py

```diff
@@ -239,7 +240,7 @@ def search():
 def config():
     if request.method == 'GET':
         return json.dumps(g.user_config.__dict__)
-    elif request.method == 'PUT':
+    elif request.method == 'PUT' and os.getenv('WHOOGLE_CONFIG_DISABLE_CHANGE', '') == '':
```

Comment:
Would be better as

```python
elif request.method == 'PUT' and not  <...>  # config disabled condition
```

rather than just comparing to an empty string. Also the "disabled config" setting should be moved up to the top of the function definition as a single variable and reused here and below in the `elif` to minimize the amount of work needed for any future refactoring of the property.